scripts/migration/GPII-4014/verify.js

```diff
@@ -0,0 +1,163 @@
+        responseErrMsg: "Error retrieving documents from database: "
+    };
+    return gpii.dbRequest.configureStep(details, options);
+};
+
+/**
+ * Given all the documents from the database, update their data according to the new data model.
+ * @param {String} responseString - the response from the request to get all documents.
+ * @param {Object} options - Where to store the to-be-updated documents:
+ * @param {Array} options.allDocs - Accumulated documents.
+ */
+gpii.migration.GPII4014.verifyDocsData = function (responseString, options) {
+    var allDocs = JSON.parse(responseString);
+    options.totalNumOfDocs = allDocs.total_rows;
+    var numOfVerified = 0;
+    var numOfErrorDocs = 0;
+    if (allDocs.rows) {
+        fluid.each(allDocs.rows, function (aRow) {
+            var hasError = false;
+            var aDoc = aRow.doc;
+            // To filter out the "_design/views" doc that doesn't have the "schemaVersion" field
+            if (aDoc.schemaVersion) {
+                numOfVerified++;
+                if (aDoc.schemaVersion !== gpii.migration.GPII4014.newSchemaVersion) {
+                    console.log("Error with the document _id \"" + aDoc._id + "\": schema version is ", aDoc.schemaVersion, ", not ", gpii.migration.GPII4014.newSchemaVersion);
+                    hasError = true;
+                }
+                if (aDoc.timestampUpdated === null) {
+                    console.log("Error with the document _id \"" + aDoc._id + "\": the value of timestampUpdated is empty");
+                    hasError = true;
+                }
+                if (aDoc.type === "clientCredential") {
+                    var docFields = {
+                        allowedIPBlocks: aDoc.allowedIPBlocks,
+                        allowedPrefsToWrite: aDoc.allowedPrefsToWrite,
+                        isCreateGpiiKeyAllowed: aDoc.isCreateGpiiKeyAllowed,
+                        isCreatePrefsSafeAllowed: aDoc.isCreatePrefsSafeAllowed
+                    };
+                    var diffResult = fluid.model.diff(
+                        docFields,
+                        options.novaClientCredentials.includes(aDoc._id) ? gpii.migration.GPII4014.newValuesForNovaClientCredential : gpii.migration.GPII4014.newValuesForPublicClientCredential
+                    );
+                    if (!diffResult) {
+                        console.log("Error with the document _id \"" + aDoc._id + "\": new field values for client credential are incorrect - ", docFields);
+                        hasError = true;
+                    }
+                }
+            }
+            if (hasError) {
+                numOfErrorDocs++;
+            }
+        });
+    }
+    console.log("The database has " + options.totalNumOfDocs + " documents. " + numOfVerified + " documents are qualified for verifications.");
+    if (numOfErrorDocs > 0) {
+        console.log("Fail: " + numOfErrorDocs + " documents have errors.");
```

Comment:
As above comment, it appears there is no route for this failure status to result in a top-level process failure